pkg/sypgp/sypgp_test.go

```diff
@@ -24,6 +29,168 @@ var (
 	testEntity *openpgp.Entity
 )
 
+type mockPKSLookup struct {
+	code int
+	el   openpgp.EntityList
+}
+
+func (ms *mockPKSLookup) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(ms.code)
+	if ms.code == http.StatusOK {
+		w.Header().Set("Content-Type", "application/pgp-keys")
+
+		wr, err := armor.Encode(w, openpgp.PublicKeyType, nil)
+		if err != nil {
+			log.Fatalf("failed to get encoder: %v", err)
+		}
+		defer wr.Close()
+
+		for _, e := range ms.el {
+			if err = e.Serialize(wr); err != nil {
+				log.Fatalf("failed to serialize entity: %v", err)
+			}
+		}
+	}
+}
+
+func TestSearchPubkey(t *testing.T) {
+	ms := &mockPKSLookup{}
+	srv := httptest.NewServer(ms)
+	defer srv.Close()
+
+	tests := []struct {
+		name      string
+		code      int
+		el        openpgp.EntityList
+		search    string
+		uri       string
+		authToken string
+		wantErr   bool
+	}{
+		{"Success", http.StatusOK, openpgp.EntityList{testEntity}, "search", srv.URL, "", false},
+		{"SuccessToken", http.StatusOK, openpgp.EntityList{testEntity}, "search", srv.URL, "token", false},
+		{"Unauthorized", http.StatusUnauthorized, nil, "search", srv.URL, "", true},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, test.WithoutPrivilege(func(t *testing.T) {
```

Comment:
I really would prefer that we don't use `test.WithoutPrivilege` or `test.WithPrivilege` functions unless we absolutely *have* to call those. That package causes problems with running tests on MacOS